### PR TITLE
PRIME#1835 site reg confirmation email different for existing site

### PR DESCRIPTION
**Pull request by Paahn**
PRIME 1768+1835
- All site reg should see new checkbox
- Send a different email when user has indicated via the above checkbox that the site was active prior to registering in PRIME.
- SiteID mandatory if checkbox is checked

### CODE REVIEW

File: prime-dotnet-webapi/Configuration/Seeded Tables/EmailTemplateConfiguration.cs [hunk 1]
```diff
@@ -26,7 +26,8 @@ public override IEnumerable<EmailTemplate> SeedData
...
+                    new EmailTemplate { Id = 15, EmailType = EmailTemplateType.SiteReviewedNotification,              Template = "A PRIME Admin has reviewed the site registration for PEC/SiteID# @Model.Pec. @(!string.IsNullOrWhiteSpace(Model.Note) ? $\"The following notes were added: {Model.Note}\" : \"\")", ModifiedDate = SEEDING_DATE, CreatedTimeStamp = SEEDING_DATE, UpdatedTimeStamp = SEEDING_DATE},
+                    new EmailTemplate { Id = 16, EmailType = EmailTemplateType.SiteActiveBeforeRegistrationSubmission, Template = "Thank you for registering your site (SiteID: @Model.Pec) in PRIME. If you need to update any site information in PRIME, you may log in at any time using your mobile BC Services Card. If you have any questions, please phone 1 - 844 - 397 - 7463 or email PRIMESupport@@gov.bc.ca. Thank you.", ModifiedDate = SEEDING_DATE, CreatedTimeStamp = SEEDING_DATE, UpdatedTimeStamp = SEEDING_DATE}
```
[the remark this answers is not in the document]
Comment (Paahn) [reply prime-dotnet-webapi/Configuration/Seeded Tables/EmailTemplateConfiguration.cs:30]: yeah it was treating it as C# code without the @@ and throwing an error when creating the template.

**Review by AMiKaH — APPROVED** [review]

File: prime-angular-frontend/src/app/shared/components/forms/site-information-form/site-information-form.component.ts [hunk 2]
```diff
@@ -81,4 +82,8 @@ export class SiteInformationFormComponent implements OnInit {
       )
       .subscribe();
   }
+
+  private getSiteOperational(): void {
+    this.form.get('activeBeforeRegistration').valueChanges.subscribe();
```
Comment (johnnyl6) [inline prime-angular-frontend/src/app/shared/components/forms/site-information-form/site-information-form.component.ts:87]: seems valueChanges is subscribed but did not do anything?

Comment (Paahn) [reply prime-angular-frontend/src/app/shared/components/forms/site-information-form/site-information-form.component.ts]: removed

**Review by johnnyl6 — APPROVED** [review]
LGTM

File: prime-dotnet-webapi/Controllers/SitesController.cs [hunk 3]
```diff
@@ -820,8 +820,15 @@ public async Task<ActionResult> ApproveSite(int siteId)
...
             var updatedSite = await _siteService.ApproveSite(siteId);
-            await _emailService.SendSiteApprovedPharmaNetAdministratorAsync(site);
-            await _emailService.SendSiteApprovedSigningAuthorityAsync(site);
+            if (site.ActiveBeforeRegistration)
+            {
+                await _emailService.SendSiteActiveBeforeRegistrationAsync(site.Id, site.Organization.SigningAuthority.Email);
+            }
+            else
+            {
+                await _emailService.SendSiteApprovedPharmaNetAdministratorAsync(site);
```
Comment (james-hollinger) [inline prime-dotnet-webapi/Controllers/SitesController.cs:829]: is the intention that the pharmanet administrator email is not sent at all for these sites?

File: prime-dotnet-webapi/Services/SiteService.cs [hunk 4]
```diff
@@ -90,6 +90,8 @@ public async Task<int> UpdateSiteAsync(int siteId, SiteUpdateModel updatedSite)
         {
             var currentSite = await GetSiteAsync(siteId);
 
+            currentSite.ActiveBeforeRegistration = updatedSite.ActiveBeforeRegistration;
+
             _context.Entry(currentSite).CurrentValues.SetValues(updatedSite);
```
Comment (james-hollinger) [inline prime-dotnet-webapi/Services/SiteService.cs:95]: this line assigns all "simple" values from the updated site to the existing site, like `string`s, `int`s, `date`s, and  `bool`s. Basically, it assigns any values that are on the `Site` table bot not child objects. So line 93 is not needed,

File: prime-dotnet-webapi/Controllers/SitesController.cs (hunk 3 above)
Comment (Paahn) [reply prime-dotnet-webapi/Controllers/SitesController.cs:829]: Sarah just confirmed, we are not sending to the Pharmanet Admin.

**Review by james-hollinger — APPROVED** [review]
LGTM

File: prime-angular-frontend/src/app/shared/components/forms/site-information-form/site-information-form.component.html [hunk 5]
```diff
@@ -57,7 +57,7 @@
           <mat-label>Site ID</mat-label>
           <input matInput
                  formControlName="pec" />
-          <mat-error *ngIf="pec.hasError('required')">
+          <mat-error *ngIf="pec.hasError('required') || activeBeforeReg.checked">
```
Comment (mtpultz) [inline prime-angular-frontend/src/app/shared/components/forms/site-information-form/site-information-form.component.html:60]: With this being a reactive form you will want to use a FormGroup validator apply validation to the PEC when activeBeforeReg is checked.

Comment (Paahn) [reply prime-angular-frontend/src/app/shared/components/forms/site-information-form/site-information-form.component.html:60]: As far as I can tell PEC is already always required

**Review by mtpultz — APPROVED** [review]
LGTM

**Pull request merged**